sound/soc/sof/sof-client.c

```diff
@@ -404,14 +409,17 @@ int sof_client_register_ipc_rx_handler(struct sof_client_dev *cdev,
 	if (!callback)
 		return -EINVAL;
 
-	if (cdev->sdev->pdata->ipc_type == SOF_IPC &&
-	    !(ipc_msg_type & SOF_GLB_TYPE_MASK))
-		return -EINVAL;
-
-	if (cdev->sdev->pdata->ipc_type == SOF_INTEL_IPC4 &&
-	    !(ipc_msg_type & (SOF_IPC4_NOTIFICATION_TYPE_MASK >>
-			      SOF_IPC4_NOTIFICATION_TYPE_SHIFT)))
+	if (cdev->sdev->pdata->ipc_type == SOF_IPC) {
+		if (!(ipc_msg_type & SOF_GLB_TYPE_MASK))
+			return -EINVAL;
+	} else if (cdev->sdev->pdata->ipc_type == SOF_INTEL_IPC4) {
+		if (!(ipc_msg_type & SOF_IPC4_NOTIFICATION_TYPE_MASK))
+			return -EINVAL;
+	} else {
+		dev_warn(sdev->dev, "%s: Not supported IPC version: %d\n",
```

Comment:
IPC version not supported?

Reply:
For me the printout:
```
sof_client_register_ipc_rx_handler: Not supported IPC version: 2
```
reads better than
```
sof_client_register_ipc_rx_handler: IPC version not supported: 2
```
But I don't have strong attachment to the former.

Reply:
I'd prefer IPC version 2 not supported :) but of course I dont care really. All of them give the same message